### PR TITLE
Slow scheduler & functions

### PR DESCRIPTION
**Pull request by Ollieollieolllie**
added a slow scheduled array 
and functionalized leftover scripts from CBA compat
moved filters into the relevant functions to stop delay from filter update caused by having variables updated slower

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreslowScheduler.sqf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* ----------------------------------------------------------------------------
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreslowScheduler.sqf:1]: Could it be possible to have just one scheduler function with two function `queues` that it worries about?  Rather than having this as two functions

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fn_loops.sqf [hunk 2]
```diff
@@ -1,3 +1,18 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_loops
+
+Description:
+
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fn_loops.sqf:5]: Needs a description, it looks like it plays the APU noise

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_Ihadsstoggle.sqf [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_Ihadsstoggle
+Description:
+Parameters:
+Returns:
+    Nothing
+    
+Examples:
+    [_heli] spawn fza_fnc_Ihadsstoggle;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_Ihadsstoggle.sqf:9]: This code doesn't have any sleeps, it is callable. There are a few more functions like this so go through and if it doesn't have any long loops / sleeps / waitUntils then use `call` rather than `spawn`

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf [hunk 4]
```diff
@@ -709,6 +709,7 @@ if (currentweapon _heli == "Laserdesignator_mounted") then {
 
 //CSCOPE
 
+_targetsToDraw = ([_heli, fza_ah64_Cscopelist] call fza_fnc_targetingFilterType);
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf:712]: This looks unrelated, make sure you mention this in the description too.

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
Good job on the performance fix, I have a few issues for you though:

* You need to add descriptions to the things you've functionised. I know they didn't have functions before, but the way I was trying to track it was anything I put into a function I read through and documented, If they don't have a return type, just say it's `Nothing` and for the author just put `Unknown`.
* Some minor capitalisation issues, I have mentioned them there.
* One performance benefit we can get off moving from `execvm` is that we can `call` those functions rather than `spawn`ing them. I would recommend checking what you can do that to.

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreslowScheduler.sqf (hunk 1 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreslowScheduler.sqf:1]: I would not know how to do such marvelous coding ideas

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf [hunk 5]
```diff
@@ -125,8 +128,5 @@ fza_ah64_ada_units = ["vme_PLA_DK9","VME_PLA_HQ64","VME_PLA_HQ7","VME_PLA_PGZ04"
...
 "B_APC_Tracked_01_AA_F","B_T_APC_Tracked_01_AA_F","RHS_Ural_Zu23_Base","UK3CB_MTLB_Zu23","UK3CB_V3S_Zu23","UK3CB_Hilux_ZU23","LT_01_scout_base_F"];
 
-
-//EXPERIMENTAL - RUN ONCE FOR PLAYER ONLY
-
-_ufdtracker = [player] execvm "\fza_ah64_controls\scripting\ufd.sqf";
-_targetscanner = [] execvm "\fza_ah64_controls\scripting\fcr_longbow.sqf";
\ No newline at end of file
+[0] spawn fza_fnc_fcrlongbow;
+[0] spawn fza_fnc_Ufd;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:132]: Lower case U

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf [hunk 6]
```diff
@@ -116,7 +118,8 @@ fza_ah64_headelev = 0;
 fza_ah64_cmpressed = 0;
 fza_ah64_nohelpers = 1;
 fza_ah64_hducolor = [0.1, 1, 0, 1];
-fza_ah64_schedarray = [fza_fnc_weaponPylonCheckValid, fza_fnc_weaponTurretAim, fza_fnc_targetingPNVSControl, fza_fnc_targetingSched, fza_fnc_avionicsSlipIndicator, fza_fnc_navigationWaypointEta, fza_fnc_ihadssDraw, fza_fnc_targetingUpdate, fza_fnc_engineGovernor, fza_fnc_mpdUpdateDisplays]; //disabled fza_ah64_cpg_controls//
+fza_ah64_schedarray = [fza_fnc_weaponTurretAim, fza_fnc_targetingPNVSControl, fza_fnc_targetingSched, fza_fnc_avionicsSlipIndicator, fza_fnc_navigationWaypointEta, fza_fnc_ihadssDraw, fza_fnc_targetingUpdate, fza_fnc_engineGovernor, fza_fnc_mpdUpdateDisplays]; //disabled fza_ah64_cpg_controls//
+fza_ah64_Slowschedarray = [fza_fnc_targetingVariable, fza_fnc_targetingUpdate, fza_fnc_weaponPylonCheckValid];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:122]: Lower case s on `fza_ah64_slowschedarray`

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fn_loops.sqf [hunk 7]
```diff
@@ -1,3 +1,20 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_loops
+
+Description:
+    loos battery or apu audio
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fn_loops.sqf:5]: loops

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssHandleControl.sqf [hunk 8]
```diff
@@ -37,10 +37,10 @@ switch(_control) do {
 		} else {
 			_heli animate["cpg_ihadss_brt", 0];
 		};
-		[_heli] execVM "\fza_ah64_controls\scripting\ihadss.sqf";
+		[_heli] call fza_fnc_ihadsstoggle;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssHandleControl.sqf:40]: `ihadssToggle` and `monocleToggle`

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/pfz/fn_pfz.sqf [hunk 9]
```diff
@@ -1,3 +1,19 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_pfz
+
+Description:
+    Main pfz function
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/pfz/fn_pfz.sqf:5]: What does this mean?

This code actually handles the creation of PFZs. Could rename to `fza_fnc_pfzCreate`.

Good description would be

> Runs the in-map user interface that is used for selecting PFZs. Generates the PFZ for the given helicopter and puts all relevant targets inside it.

You also need to fill out the parameters section, e.g.

```sqf
_heli - the helicopter to create the PFZ on
_pfznum - the index of the PFZ (in range 1 - 8 inclusive)

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/pfz/fn_pfzReceive.sqf [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_pfzReceive
+
+Description:
+    sends out the target data
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/pfz/fn_pfzReceive.sqf:5]: You have the send and receive descriptions the wrong way round. I would also recommend being more specific on what it is doing - broadcasting out the currently selected PFZ for other helis or receiving that PFZ

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

**Review by mattysmith22 — APPROVED** [review]
